chain/chain/src/store.rs

```diff
@@ -991,4 +997,39 @@ impl<'a, T: ChainStoreAccess> ChainStoreUpdate<'a, T> {
         let store_update = self.finalize()?;
         store_update.commit().map_err(|e| e.into())
     }
+
+    pub fn check_block_validity(&mut self, block: &Block) -> Result<(), Error> {
```

Comment:
why is this in the store? why not just put it on the Block actually?

Reply:
awesome idea, thank you